src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/GetAzureCognitiveServicesAccount.cs

```diff
@@ -53,46 +62,86 @@ public class GetAzureCognitiveServicesAccountCommand : CognitiveServicesAccountB
+                        WriteCognitiveServicesAccountList(deletedAccounts);
+                        break;
+                    default:
+                        if (string.IsNullOrEmpty(this.ResourceGroupName))
+                        {
+                            var cognitiveServicesAccounts = GetWithPaging(this.CognitiveServicesClient.Accounts.List(), false);
 
-                    WriteCognitiveServicesAccountList(cognitiveServicesAccounts);
-                }
-                else if (string.IsNullOrEmpty(this.Name))
-                {
-                    var cognitiveServicesAccounts = GetWithPaging(this.CognitiveServicesClient.Accounts.ListByResourceGroup(this.ResourceGroupName), true);
-                    if (cognitiveServicesAccounts == null)
-                    {
-                        WriteWarningWithTimestamp("Received empty accounts list");
-                    }
-                    WriteCognitiveServicesAccountList(cognitiveServicesAccounts);
-                }
-                else
-                {
-                    var cognitiveServicesAccount = this.CognitiveServicesClient.Accounts.GetProperties(
-                        this.ResourceGroupName,
-                        this.Name);
+                            WriteCognitiveServicesAccountList(cognitiveServicesAccounts);
+                        }
+                        else if (string.IsNullOrEmpty(this.Name))
+                        {
+                            var cognitiveServicesAccounts = GetWithPaging(this.CognitiveServicesClient.Accounts.ListByResourceGroup(this.ResourceGroupName), true);
+                            if (cognitiveServicesAccounts == null)
+                            {
+                                WriteWarningWithTimestamp("Received empty accounts list");
+                            }
+                            WriteCognitiveServicesAccountList(cognitiveServicesAccounts);
+                        }
+                        else
+                        {
+                            var cognitiveServicesAccount = this.CognitiveServicesClient.Accounts.Get(
+                                this.ResourceGroupName,
+                                this.Name);
 
-                    WriteCognitiveServicesAccount(cognitiveServicesAccount);
+                            WriteCognitiveServicesAccount(cognitiveServicesAccount);
+                        }
+                        break;
                 }
+
             });
         }
 
-        private IEnumerable<CognitiveServicesAccount> GetWithPaging(IPage<CognitiveServicesAccount> firstPage, bool isResourceGroup)
+        private IEnumerable<Account> GetWithPaging(IPage<Account> firstPage, bool isResourceGroup)
         {
-            var cognitiveServicesAccounts = new List<CognitiveServicesAccount>(firstPage);
-            IPage<CognitiveServicesAccount> nextPage = null;
+            var cognitiveServicesAccounts = new List<Account>(firstPage);
```

Comment:
You should return the type the same as what you defined as cmdlet output.

Reply:
Yes, `WriteCognitiveServicesAccount` and `WriteCognitiveServicesAccountList` will convert this to `PSCognitiveServicesAccount`